tests/unit/test_structured_config.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Structured config unit tests."""
+
+import logging
+
+import pytest
+from ops.testing import Harness
+
+from charm import AirbyteK8SOperatorCharm
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def _harness():
```

Comment:
Minor:

there are special rules for pytest fixtures.
because of that, there isn't really a "private" method notation.

```suggestion
def harness():
```

P.S. a fixture with the same name overrides a fixture from an outer scope, so even if there is a `harness` fixture somewhere, a custom name is not called for.